### PR TITLE
add cmdline

### PR DESCRIPTION
**Pull request by v1nh1shungry**
add https://github.com/tanakh/cmdline

### CODE REVIEW

File: packages/c/cmdline/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    add_urls("https://github.com/tanakh/cmdline.git")
+    add_versions("2014.2.4", "e4cd007fb8f0314002d9a5b4d82939106e4144e4")
+
+    on_install(function (package)
+      os.cp("cmdline.h", package:installdir("include"))
+    end)
+
+    on_test(function (package)
+      assert(package:has_cxxincludes("cmdline.h"))
```
Comment (waruqi) [inline packages/c/cmdline/xmake.lua:16]: space4 indent

File: packages/c/cmdline/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    add_urls("https://github.com/tanakh/cmdline.git")
+    add_versions("2014.2.4", "e4cd007fb8f0314002d9a5b4d82939106e4144e4")
+
+    on_install(function (package)
+        os.cp("cmdline.h", package:installdir("include"))
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxincludes("cmdline.h"))
```
Comment (waruqi) [inline packages/c/cmdline/xmake.lua:16]: please use has_cxxfuncs/has_cfuncs to check a interface function.

**Comment by waruqi** [comment]
> C:\Users\runneradmin\AppData\Local\.xmake\packages\c\cmdline\2014.2.4\4c92bae95dee4029[82](https://github.com/xmake-io/xmake-repo/runs/7553232766?check_suite_focus=true#step:5:83)7547582358a178\include\cmdline.h(39): fatal error C1083: Cannot open include file: 'cxxabi.h': No such file or directory

on windows.

**Comment by v1nh1shungry** [comment]
Yes, I notice it. I just realized that cmdline doesn't support msvc without a patch (https://github.com/tanakh/cmdline/issues/6) cause I don't have msvc installed.


> > C:\Users\runneradmin\AppData\Local.xmake\packages\c\cmdline\2014.2.4\4c92bae95dee4029[82](https://github.com/xmake-io/xmake-repo/runs/7553232766?check_suite_focus=true#step:5:83)7547582358a178\include\cmdline.h(39): fatal error C1083: Cannot open include file: 'cxxabi.h': No such file or directory
> 
> on windows.
> 
> If it does not support windows, please disable it.
> 
> ```lua
> on_install("linux", "macosx", "bsd", function(package)
> end)
> ```

Thanks! I will fix it right now.

**Comment by waruqi** [comment]
you can also use `add_patches` to add a patch for msvc.

**Comment by v1nh1shungry** [comment]
> you can also use `add_patches` to add a patch for msvc.

Thanks for reminding. I may look into this later in another PR since I now don't use msvc and don't have it installed.

**Pull request merged**